lib/total.rb

```diff
@@ -0,0 +1,27 @@
+require_relative 'checkout'
+
+class Total
+  attr_reader :selection, :checkout
+
+  def initialize(checkout = Checkout.new)
+    @selection = []
+    @checkout = checkout
+  end
+
+  def bill
+    sum = []
+    selection.each do |item|
+      sum << item[:price].to_f * item[:quantity]
+    end
+    sum.inject(:+)
+  end
+
+  def receipt
+    selection
+  # not entirely sure what the user story required

```

Comment:
It's probably fine to just leave out the `.selection` method to get the order list — if we wanted to have some printed and formatted receipt, it would probably become the job of another class (since this would deal with presenting/displaying data to the user, which is a different concern)